detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmYamlTransformer.java

```diff
@@ -114,18 +126,41 @@ private String convertRawEntryToPackageId(Map.Entry<String, String> entry, PnpmL
         return String.format("/%s/%s", name, version);
     }
 
-    private NameVersion parseNameVersionFromId(String id) {
+    private Optional<NameVersion> parseNameVersionFromId(String id) {
         // ids follow format: /name/version, where name often contains slashes
-        int indexOfLastSlash = id.lastIndexOf("/");
-        String name = id.substring(1, indexOfLastSlash);
-        String version = id.substring(indexOfLastSlash + 1);
+        try {
+            int indexOfLastSlash = id.lastIndexOf("/");
+            String name = id.substring(1, indexOfLastSlash);
+            String version = id.substring(indexOfLastSlash + 1);
+            return Optional.of(new NameVersion(name, version));
+        } catch (Exception e) {
+            logger.debug(String.format("There was an issue parsing package id: %s.  This is likely an unsupported format.", id));
+            return Optional.empty();
+        }
+    }
 
-        return new NameVersion(name, version);
+    private Optional<Dependency> buildDependencyFromPackageEntry(Map.Entry<String, PnpmPackageInfo> packageEntry) {
+        PnpmPackageInfo packageInfo = packageEntry.getValue();
+        if (packageInfo.name != null && packageInfo.version != null) {
+            return Optional.of(new Dependency(externalIdFactory.createNameVersionExternalId(Forge.NPMJS, packageInfo.name, packageInfo.version)));
+        }
+        return buildDependencyFromPackageId(packageEntry.getKey());
+    }
+
+    private Optional<Dependency> buildDependencyFromPackageId(String packageId) {
+        Optional<NameVersion> nameVersion = parseNameVersionFromId(packageId);
+        if (nameVersion.isPresent()) {
+            return Optional.of(new Dependency(externalIdFactory.createNameVersionExternalId(Forge.NPMJS, nameVersion.get().getName(), nameVersion.get().getVersion())));
+        }
+        return Optional.empty();
     }
 
-    private Dependency buildDependencyFromPackageId(String packageId) {
-        NameVersion nameVersion = parseNameVersionFromId(packageId);
-        return new Dependency(externalIdFactory.createNameVersionExternalId(Forge.NPMJS, nameVersion.getName(), nameVersion.getVersion()));
+    private boolean isRootPackage(String id, List<String> rootIds) {
+        return rootIds.contains(id) ||
+            rootIds.stream()
+                .map(this::parseNameVersionFromId)
+                .filter(Optional::isPresent)
+                .anyMatch(nameVersion -> nameVersion.get().getVersion().equals(id)); // for file dependencies, they are declared as <name> : <fileIdAsReportedInPackagesSection>
```

Comment:
I would take the stream a little bit farther.
```suggestion
                .filter(Optional::isPresent)
                .map(Optional::get)
                .map(NameVersion::getVersion)
                .anyMatch(id::equals); // for file dependencies, they are declared as <name> : <fileIdAsReportedInPackagesSection>
```